internal/fetchclient/fetchclient.go

```diff
@@ -41,21 +41,27 @@ type Client struct {
 }
 
 // New returns a new client.
-func New(ctx context.Context) *Client {
-	var client *http.Client
-	if ghToken := os.Getenv("GITHUB_TOKEN"); ghToken != "" {
-		ts := oauth2.StaticTokenSource(
-			&oauth2.Token{AccessToken: ghToken},
-		)
-		client = oauth2.NewClient(ctx, ts)
+func New(ctx context.Context) (context.Context, *Client) {
+	githubToken := os.Getenv("GITHUB_TOKEN")
+	if githubToken != "" {
+		log.Printf("creating authenticated GitHub client with GITHUB_TOKEN")
 	} else {
-		retryableClient := retryablehttp.NewClient()
-		retryableClient.Logger = nil
-		client = retryableClient.StandardClient()
+		log.Printf("creating unauthenticated GitHub client")
 	}
-	return &Client{
-		httpClient: client,
-		ghClient:   github.NewClient(client),
+	httpClient, err := github_ratelimit.NewRateLimitWaiterClient(nil)
+	if err != nil {
+		log.Printf("failed to create rate limiter: %v", err)
+		// Fallback to default client if rate limiter creation fails.
+		ctx = context.WithValue(ctx, github.SleepUntilPrimaryRateLimitResetWhenRateLimited, true)
+		httpClient = http.DefaultClient
+	} else {
+		// Disable the github-go rate limiter for github_ratelimit.
+		ctx = context.WithValue(ctx, github.BypassRateLimitCheck, true)
+	}
+	githubClient := github.NewClient(httpClient).WithAuthToken(githubToken)
+	return ctx, &Client{
+		httpClient: httpClient,
```

Comment:
Need to validate this `httpClient` doesn't require oauth2 tokens.

Reply:
`httpClient` looks to only be called for fetching against other registries. Should _not_ have the token `GITHUB_TOKEN` set.